js/commerce_bluesnap.form.js

```diff
@@ -0,0 +1,252 @@
+/**
+ * @file
+ * Javascript to submit card details to BlueSnap in a PCI-compliant way.
+ */
+
+(function ($, Drupal, drupalSettings) {
+
+  'use strict';
+
+  /**
+   * Attaches the commerceBluesnapForm behavior.
+   *
+   * @type {Drupal~behavior}
+   *
+   * @prop {Drupal~behaviorAttach} attach
+   *   Attaches the commerceBluesnapForm behavior.
+   * @prop {Drupal~behaviorDetach} detach
+   *   Detaches the commerceBluesnapForm behavior.
+   *
+   * @see Drupal.commerceBluesnap
+   */
+  Drupal.behaviors.commerceBluesnapForm = {
+    attach: function (context) {
+      $('.bluesnap-form', context).once('bluesnap-processed').each(function () {
+        // Initialize the card field variables.
+        var isCardNumberComplete = false;
+        var isCVVComplete = false;
+        var isExpiryComplete = false;
+
+        // Get the form.
+        var $form = $(this).closest('form');
+        // Get the Hosted Payment Fields token.
+        var $hostedPaymentFieldsToken = drupalSettings.bluesnap_token;
+
+        /**
+         * Takes token and bsObj as inputs and initiates Hosted Payment Fields.
+         */
+        $(document).ready(function () {
```

Comment:
Looking a bit into it, it seems that Drupal attaches behaviors after the document is ready. Even though as I have mentioned I have had cases in the past where I had to manually do that, like here, to get the expected behavior. It might be worth validating whether we actually need `$(document).ready` or not.

Reply:
I've tested this again and can confirm that we don't.